### PR TITLE
Allow customizing redirect path for users with `shouldResetPassword` == true

### PR DESCRIPTION
**Pull request by siemensikkema**

### CODE REVIEW

File: Sources/AdminPanelProvider/Middlewares/ProtectMiddleware.swift [hunk 1]
```diff
@@ -6,10 +6,15 @@ public typealias ProtectMiddleware = CustomUserProtectMiddleware<AdminPanelUser>
 
 /// Redirects unauthenticated requests to a supplied path.
 public final class CustomUserProtectMiddleware<U: AdminPanelUserType>: Middleware {
+    let passwordEditPathForUser: (U) -> String
```
Comment (steffendsommer) [inline Sources/AdminPanelProvider/Middlewares/ProtectMiddleware.swift:9]: Do we need something default here to avoid the breaking change?

**Review by steffendsommer — CHANGES_REQUESTED** [review]
One comment about if we should avoid making this a breaking change - also, are you up for fixing the breaking tests?

File: Sources/AdminPanelProvider/Middlewares/ProtectMiddleware.swift (hunk 1 above)
Comment (siemensikkema) [reply Sources/AdminPanelProvider/Middlewares/ProtectMiddleware.swift:9]: I didn't consider that this Middleware was public. I can put in the same value as the provider gets.

**Review by steffendsommer — APPROVED** [review]

**Comment by siemensikkema** [comment]
@steffendsommer should we tag it with 0.7.1 or 0.8.0?

**Comment by steffendsommer** [comment]
@siemensikkema since it's non-breaking I would suggest 0.7.1

**Pull request merged**